cfme/automate/dialog_element.py

```diff
@@ -186,11 +130,60 @@ def reorder_elements(self, add_element, second_element, element_data):
         view.flash.assert_no_error()
 
 
+@attr.s
+class ElementCollection(BaseCollection):
+
+    ENTITY = Element
+
+    @property
+    def tree_path(self):
+        return self.parent.tree_path
+
+    def create(self, element_data=None):
+        for element in element_data:
+            view = navigate_to(self, "Add")
+            if(view.ele_label.value != ""):
+                view.plus_btn.item_select("Add a new Element to this Box")
+            view.fill(element)
+            self.set_element_type(view, element)
+        view.add_button.click()
+        view.flash.assert_no_error()
+        view.flash.assert_message('Dialog "{}" was added'.
+            format(self.parent.tab.dialog.label))
+        view.flash.assert_no_error()
+        return self.instantiate(element_data=element_data)
+
+    def set_element_type(self, view, element):
+        """ Method to add element type.Depending on their type the subfields varies.
+
+        Args:
+            each_element: subfields depending on element type.
+        """
+        choose_type = element.get("choose_type")
+        dynamic_chkbox = element.get("dynamic_chkbox")
+        element_type = ['Drop Down List', 'Radio Button']
+        if choose_type in element_type:
+            if not dynamic_chkbox:
+                row = view.entry_table.row(Value='<New Entry>')
+                row.click()
+                view.fill({'entry_value': "Yes",
+                           'entry_description': "entry_desc"})
+                view.add_entry_button.click()
+            else:
+                node1 = "InspectMe"
+                view.fill({'field_entry_point': 'b'})
+                view.bt_tree.click_path("Datastore", "new_domain", "System", "Request", node1)
+                view.apply_btn.click()
+                view.fill({'field_show_refresh_button': True})
+        if choose_type == "Text Area Box":
+            view.fill({'text_area': 'Default text'})
+
+
 @navigator.register(ElementCollection)
 class Add(CFMENavigateStep):
     VIEW = AddElementView
 
-    prerequisite = NavigateToAttribute('parent.collection', 'Add')
+    prerequisite = NavigateToAttribute('parent.parent', 'Add')
```

Comment:
@psav thought - we might want a `NavigateToParent/NavigateToParentOfType`

Reply:
Yeh I thought about that too